### PR TITLE
[SNC-232] impove test diff

### PR DESCRIPTION
**Pull request by davidmdm**
## Rationale

The CLI test output is confusing. It is unclear what is the expected vs actual output of a test diff. 
The test data is also defined in yaml but the diff comes out in JSON format.

## Considerations

Use a well-known and used diff library! So I selected the same one that powers `testify/assert`.

### Before 
image

### After
image

### CODE REVIEW

File: cpa/tester/runner.go [hunk 1]
```diff
@@ -13,8 +13,9 @@ import (
 	"github.com/CircleCI-Public/circle-policy-agent/cpa"
 	"github.com/CircleCI-Public/circle-policy-agent/internal"
 	"github.com/open-policy-agent/opa/tester"
-	"github.com/yazgazan/jaydiff/diff"
+	"github.com/pmezard/go-difflib/difflib"
```
Comment (eddiewebb) [inline cpa/tester/runner.go:16]: does this library perhaps also change the consideration of ordered lists? showing the same item as expexted and actual, but erroneous?


This example was recent: rule ` rule_reserved_cera_contexts` is expected, but came after the unexpected `rule_contexts_allowed_by_project_ids`
https://app.circleci.com/pipelines/github/AwesomeCICD/circleci-policies/349/workflows/b8ac42fe-3a4e-4fc0-aeaf-cd433504bf89/jobs/392/tests#failed-test-0

```
{
     "enabled_rules": [
-      "rule_reserved_cera_contexts",
+      "rule_contexts_allowed_by_project_ids",
+      "rule_reserved_cera_contexts"
     ],
     "status": "PASS"
   }
```

Ideally this is more like:


```
{
     "enabled_rules": [
+      "rule_contexts_allowed_by_project_ids",
     ],
     "status": "PASS"
   }
```

**Review by eddiewebb — APPROVED** [review]
New output is easier to consume as it does not require the same mental mapping I needed to before. 

🎉

File: cpa/tester/runner.go (hunk 1 above)
Comment (davidmdm) [reply cpa/tester/runner.go:16]: I dont know if this answers your question but I get output like so for arrays:
```
FAIL    test_snyk/not_ccc_job    0.000s    

    --- Expected
    +++ Actual
    @@ -1,2 +1,3 @@
     enabled_rules:
    +    - check_snyk_orb
         - enforce_snyk_scan
 ```

**Review by rooneyshuman — APPROVED** [review]
LGTM!

**Review by sagar-connect — APPROVED** [review]

**Pull request merged**